src/blockly-blocks/lava/block-simulate-lava.js

```diff
@@ -157,3 +181,22 @@ Blockly.JavaScript.molasses_run_simulation = function(block) {
   ${contents}
   this.runMolassesSimulation();\n`;
 };
+
+Blockly.JavaScript.molasses_set_flag_location = function(block) {
+  const flagName = block.getFieldValue("name") ?? "";
+  if (flagName && flagName.length > 15) {
+    block.setWarningText("Flag name cannot be more than 15 characters.");
+    return "";
+  }
+  const flagColor = block.getFieldValue("color") || "green";
```

Comment:
```suggestion
  const flagColor = block.getFieldValue("color") || flagColors[0];
```
In case they change the set of colors or their order.